NumLib/DOF/LocalToGlobalIndexMap.cpp

```diff
@@ -53,27 +53,39 @@ LocalToGlobalIndexMap::LocalToGlobalIndexMap(
     // For all MeshSubsets and each of their MeshSubset's and each element
     // of that MeshSubset save a line of global indices.
 
-    unsigned comp_id = 0;
-    for (auto const& mss : _mesh_subsets)
+
+    _variable_component_offsets.reserve(_mesh_subsets.size());
+    std::size_t offset = 0;
+    for (int variable_id = 0; variable_id < _mesh_subsets.size();
+         ++variable_id)
     {
-        for (MeshLib::MeshSubset const* const ms : *mss)
+        _variable_component_offsets.push_back(offset);
+        auto const& mss = *_mesh_subsets[variable_id];
+        for (int component_id = 0; component_id < mss.size();
+             ++component_id)
         {
-            std::size_t const mesh_id = ms->getMeshID();
+            MeshLib::MeshSubset const& ms = mss.getMeshSubset(component_id);
+            std::size_t const mesh_id = ms.getMeshID();
 
-            findGlobalIndices(ms->elementsBegin(), ms->elementsEnd(), mesh_id,
-                              comp_id, comp_id);
+            findGlobalIndices(
+                ms.elementsBegin(), ms.elementsEnd(), mesh_id,
+                getGlobalComponent(variable_id, component_id),
+                getGlobalComponent(variable_id, component_id));
```

Comment:
It's not expensive (around two additions and fetching memory once). Removed the duplication anyway.